odml/base.py

```diff
@@ -136,11 +136,17 @@ def __contains__(self, key):
             if (hasattr(obj, "name") and obj.name == key) or key == obj:
                 return True
 
-    def append(self, obj):
-        if obj.name in self:
-            raise KeyError("Object with the same name already exists! " + str(obj))
-        else:
-            super(SmartList, self).append(obj)
+    def append(self, *obj_tuple):
+        ll = len(obj_tuple)
+        for i in list(range(0,ll)):
+                if obj_tuple[i].name in self:
+                    raise KeyError("Object with the same name already exists! " + str(obj_tuple[i]))
+                from odml.section import BaseSection
+                from odml.doc import BaseDocument
```

Comment:
Is there a specific reason for importing within the loop? Or, the other way round why not importing on a global scope? Again, methinks this reduces performance, but maybe python optimises it away.

My suggestion here: make global import and be done with it